index.bs

```diff
@@ -2846,6 +2893,10 @@ The following {{DOMException}} exceptions can be raised:
     ::  The ceremony was cancelled by an {{AbortController}}.
         See [[#sctn-abortoperation]] and [[#sctn-sample-aborting]].
 
+    :   {{HybridPrerequisitesError}}
```

Comment:
I don't really understand where these hybrid errors would be used. There is no way for an RP to mandate a specific transport without an allow list, and even with an allowlist, all credential IDs with hybrid would also have internal.

We also don't provide this level of granularity for other transports. What if I don't have an NFC reader? What if it is the Bluetooth transport and Bluetooth is not available?

IMO, this level of detail should not be disclosed to the RP. Is the WebAuthn client's responsibility to remediate transport issues and/or guide the user to an alternative option.

Hybrid availability is also advertised in `getClientCapabilities()`.

Reply:
> I don't really understand where these hybrid errors would be used.

I tried to capture my thinking [in the explainer](https://github.com/w3c/webauthn/wiki/Explainer:-New-Error-Codes-(2024-Edition)#hybridprerequisiteserror):

> The browser wanted to prompt the user to try for a hybrid authentication, but found that it lacked OS-level permission to use Bluetooth. After notifying the user of this, the user cancels the ceremony. The browser raises a `HybridPrerequisitesError` error, **which the Relying Party detects and shows the user how to grant the browser Bluetooth permission.** The user is able to turn on Bluetooth on their own and then successfully complete auth.

These errors can enable RPs to offer additional remediation guidance over top of whatever a user agent may wish to implement.

> There is no way for an RP to mandate a specific cross-platform transport without an allow list, and even with an allowlist, all credential IDs with hybrid would also have internal.

An assumption here is that the `"internal"` transport was a miss so the user agent proposes an attempt via `"hybrid"`.

> We also don't provide this level of granularity for other transports. What if I don't have an NFC reader? What if it is the Bluetooth transport and Bluetooth is not available?

To this I ask, could we not expose more granular errors for other transports?

Reply:
> These errors can enable RPs to offer additional remediation guidance over top of whatever a user agent may wish to implement.

This is not the RP's responsibility, and honestly if I were a user, I would be suspicious of a website asking me to change system settings.

> An assumption here is that the "internal" transport was a miss so the user agent proposes an attempt via "hybrid".

But the client has no way to know whether the request will be successful even if hybrid is possible. This error can lead the RP to believe that if the hybrid protocol issues are resolved, there will be a successful ceremony, which is not always the case. So then you end up with two failed transactions. 

> To this I ask, could we not expose more granular errors for other transports?

Because it is the WebAuthn client's responsibility to provide all of this, as it is responsible for interfacing with authenticators.

Reply:
This error is being considered for a different PR than this one now.